### PR TITLE
Return storage package to this repository.

### PR DESCRIPTION
**Pull request by marstr**
One of the conditions for moving storage into its own repository was continued support of folks consuming it out of this repository. Our mechanism for doing that was adding a git sub-module as seen here: #556. We tested moving forward to adopt the latest bits, but missed an important scenario: using glide/git to tie yourself to a commit that predated the sub-module. This caused people to stop being able to use our SDK as seen in #581.
The immediate solution was to back-out the sub-module, replacing it with copies of the files as they existed at the commit the sub-module referred to. However, we were then left with a choice. Do we permanently move to the new repository, deprecating the copy here, or deprecate the new repository and return to developing out of a single repository?
While there were supporters of both options on the team, we made the decision to return the code here based on the download numbers we saw for the new repository as seen in GitHub's Graphs. Almost all of the clones of `Azure/azure-storage-go` came indirectly through the sub-module, meaning almost everybody using that package was doing so via this repository.

For that reason, I've rebased all commits from the old repository onto our history here.

### CODE REVIEW

**Comment by marstr** [comment]
What are your thoughts, @kpfaulkner?

**Comment by kpfaulkner** [comment]
Although personally I prefer it as a separate repo, if reality is that most people using it are getting it through azure-sdk-for-go then I'd say merge it back.

**Comment by kpfaulkner** [comment]
Just to clarify, gong forward will we still be working off the dev branch I assume?

File: .travis.yml [hunk 1]
```diff
@@ -18,12 +18,14 @@ script:
...
   - test -z "$(gofmt -s -l -w management | tee /dev/stderr)"
+  - test -z "$(gofmt -s -l -w storage | tee /dev/stderr)"
   - test -z "$(gofmt -s -l -w Gododir | tee /dev/stderr)"
   - test -z "$(go build $(find ./* -type d -print | grep -v '^./vendor$' | grep -v '^./storage$') | tee /dev/stderr)"
   - test -z "$(go vet $(find ./arm/* -type d -print) | tee /dev/stderr)"
   - test -z "$(golint ./arm/... | tee /dev/stderr)"
   - go test -v ./management/...
   - test -z "$(golint ./management/... |  grep -v 'should have comment' | grep -v 'stutters' | tee /dev/stderr)"
   - go vet ./management/...
+  - go vet ./storage/...
```
Comment (mcardosos) [inline .travis.yml]: >go vet ./storage/... [](start = 4, length = 20)

At some point we have to add `go test ./storage/...`

**Comment by mcardosos** [comment]
> [GoDoc](https://godoc.org/github.com/Azure/azure-sdk-for-go) 

Why is this file inside a folder?

---
Refers to: README.md:2 in 0574446. [](commit_id = 05744467ef80e9bb8faf7159908a39f181141cdf, deletion_comment = False)

**Comment by marstr** [comment]
Yes, @kpfaulkner. The plan is to continue working off of the dev branch.

**Pull request merged**